### ISSUE TITLE
Piston not working, and using 0 ticks pistons (or a piston with a 0 tick clock) crash the game.

### ISSUE THREAD

**Issue by PrysmikDove** [QuickCarpet/QuickCarpet#150]
**Versions**
- Minecraft: ...
- QuickCarpet: ...

**Crash report**
```
Paste either the crash report (found in .minecraft/crash-reports/) or latest.log (found in .minecraft/logs/).
Be careful that you remove personal information in case you send latest.log
```

**Comment by skyrising** [comment]
The point of the issue template is that you actually fill in the requested information. Like this I can't know what's wrong.

**Comment by PrysmikDove** [comment]
Pistons are not working, when i try to use a flying machine or a 0 tick piston the game crashes.

**Comment by skyrising** [comment]
Yeah, I got that much from the title. But you're reporting a crash. Where is the crash report? Which Minecraft version? Which QuickCarpet version?
Because for me everything works fine. To help with your problem I need to know what might cause it for you.

**Comment by PrysmikDove** [comment]
Minecraft version 1.16.5, im not using any mod besides quickcarpet. Im using quickcarpet [3.0.0] 

[crash-2022-12-14_17.49.35-server.txt](https://github.com/QuickCarpet/QuickCarpet/files/10231577/crash-2022-12-14_17.49.35-server.txt)          Thats the crash report.  This is the exit code: -1 The game crashed whilst exception while updating neighbours
Error: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_2674 failed

**Comment by PrysmikDove** [comment]
Also alll /carpet settings are set to default.

**Comment by skyrising** [comment]
I think that might be caused by a change to fabric-loader after 1.16.5. Which version of that are you using?

The 1.16.5 version of QuickCarpet will receive no further updates, but there is a newer QuickCarpet version (https://github.com/QuickCarpet/QuickCarpet/releases/tag/v3.1.1) for 1.16.5, have you tried that?

**Comment by PrysmikDove** [comment]
i´ll try it

**Comment by PrysmikDove** [comment]
im using the newer quickcarpet version, but its still not working

**Comment by PrysmikDove** [comment]
hi?
